examples/src/main/kotlin/io/ethers/examples/batchrequests/SimulateBlockTransactions.kt

```diff
@@ -0,0 +1,73 @@
+package io.ethers.examples.batchrequests
+
+import io.ethers.core.isFailure
+import io.ethers.core.types.StateOverride
+import io.ethers.core.types.toBlockOverride
+import io.ethers.core.types.tracers.CallTracer
+import io.ethers.core.types.tracers.MuxTracer
+import io.ethers.core.types.tracers.PrestateTracer
+import io.ethers.core.types.tracers.TracerConfig
+import io.ethers.providers.Provider
+import kotlinx.cli.ArgParser
+import kotlinx.cli.ArgType
+import kotlinx.cli.required
+
+class SimulateBlockTransactions(
+    rpcUrl: String,
+) {
+    // Init provider
+    private val provider = Provider.fromUrl(rpcUrl).unwrap()
+
+    fun run() {
+        // For each new block, simulate all transactions in the block with traceCall
+        provider.subscribeNewHeads().sendAwait().unwrap().forEach { head ->
+            println("New block: ${head.number}")
+
+            val blockWithTransactions = provider.getBlockWithTransactions(head.number).sendAwait().unwrap()
+            val stateOverrides = StateOverride.copy(emptyMap())
+            val blockOverrides = blockWithTransactions.toBlockOverride()
+            for (tx in blockWithTransactions.transactions) {
+                val txReceipt = provider.getTransactionReceipt(tx.hash).sendAwait().unwrap().get()
+                val config = TracerConfig(
+                    tracer = MULTICALL_MUX_TRACER,
+                    stateOverrides = stateOverrides,
+                    blockOverrides = blockOverrides,
+                )
+                val unwrappedResult = provider.traceCall(tx, head.number, config).sendAwait()
```

Comment:
We should execute the call on `head.number - 1` so we're simulating on top of correct state.

And let's add an explanation that since `Transaction` (and thus `RPCTransaction`) implements `IntoCallRequest` interface, it can be passed directly into the call, without manually converting it to `CallRequest`.